src/app/main.rs

```diff
@@ -420,13 +421,29 @@ where
                     .push(buffer::layer(MAX_IN_FLIGHT))
                     .push(timeout::layer(config.bind_timeout))
                     .push(limit::layer(MAX_IN_FLIGHT))
+                    .push(strip_header::layer(super::DST_OVERRIDE_HEADER))
                     .push(router::layer(|req: &http::Request<_>| {
-                        let addr = super::http_request_authority_addr(req)
-                            .or_else(|_| super::http_request_host_addr(req))
-                            .or_else(|_| super::http_request_orig_dst_addr(req))
-                            .ok();
-                        debug!("outbound addr={:?}", addr);
-                        addr
+                        let addr = || {
+                            super::http_request_authority_addr(req)
+                                .or_else(|_| super::http_request_host_addr(req))
+                                .or_else(|_| super::http_request_orig_dst_addr(req))
+                                .ok()
+                        };
+                        super::http_request_l5d_override_dst_addr(req)
+                            .ok()
+                            .map(|override_addr| {
+                                debug!(
+                                    "outbound addr={:?}; dst-override={:?}",
+                                    addr(),
+                                    override_addr
+                                );
+                                override_addr
+                            })
+                            .or_else(|| {
+                                let addr = addr();
+                                debug!("outbound addr={:?}", addr);
+                                addr
```

Comment:
There are two places where we may need to compute `addr`. The debug line above also needs the same logic, but neither of the cases is guaranteed to happen.

Reply:
Yeah, the intention behind the closure was to avoid repeating the same code for logging the pre-override addr for debugging and for actually getting the addr when the override header is not set.

Reply:
We should not log the addr that we're not using.  Let's make `outbound addr=...` always reference the address that's being used for routing, and not contextual.

We can either do that by having a line like `outbound addr=...; dst-override`; or we can have a log line "outbound dst-override=..."; and always log the addr being used with `outbound addr=...`

Reply:
@olix0r Okay!   thought logging what the addr _would_ have been if it were not overridden would be useful for debugging purposes, but not logging it simplifies things significantly.

Reply:
Yeah, let's optimize for simplicity and ensure that we (1) log the dst addr being used and (2) indicate whether it was or was not from an override.